cli/deno_dir.rs

```diff
@@ -265,95 +264,72 @@ impl DenoDir {
     )
   }
 
-  // Prototype: https://github.com/denoland/deno/blob/golang/os.go#L56-L68
-  // TODO: this method should take deps filepath and return URL for module
-  fn src_file_to_url(self: &Self, filename: &str) -> String {
-    let filename_path = Path::new(filename);
-    if filename_path.starts_with(&self.deps) {
-      let (rest, prefix) = if filename_path.starts_with(&self.deps_https) {
-        let rest = filename_path.strip_prefix(&self.deps_https).unwrap();
-        let prefix = "https://".to_string();
-        (rest, prefix)
-      } else if filename_path.starts_with(&self.deps_http) {
-        let rest = filename_path.strip_prefix(&self.deps_http).unwrap();
-        let prefix = "http://".to_string();
-        (rest, prefix)
-      } else {
-        // TODO(kevinkassimo): change this to support other protocols than http
-        unimplemented!()
-      };
-      // Windows doesn't support ":" in filenames, so we represent port using a
-      // special string.
-      // TODO(ry) This current implementation will break on a URL that has
-      // the default port but contains "_PORT" in the path.
-      let rest = rest.to_str().unwrap().replacen("_PORT", ":", 1);
-      prefix + &rest
-    } else {
-      String::from(filename)
-    }
-  }
-
-  /// Returns (module name, local filename)
-  pub fn resolve_module_url(
+  /// This method returns local file path for given module url that is used
+  /// internally by DenoDir to reference module.
+  ///
+  /// For specifiers starting with `file://` returns the input.
+  ///
+  /// For specifier starting with `http://` and `https://` it returns
+  /// path to DenoDir dependency directory.
+  pub fn url_to_deps_path(
     self: &Self,
-    specifier: &str,
-    referrer: &str,
-  ) -> Result<Url, url::ParseError> {
-    debug!(
-      "pre-resolve_module specifier {} referrer {}",
-      specifier, referrer
-    );
-
-    let specifier = self.src_file_to_url(specifier);
-    let referrer = self.src_file_to_url(referrer);
-
-    debug!(
-      "resolve_module specifier {} referrer {}",
-      specifier, referrer
-    );
+    url: &Url,
+  ) -> Result<PathBuf, url::ParseError> {
+    let filename = match url.scheme() {
+      "file" => url.to_file_path().unwrap(),
+      "https" => get_cache_filename(self.deps_https.as_path(), &url),
+      "http" => get_cache_filename(self.deps_http.as_path(), &url),
+      scheme => {
+        eprintln!("Unsupported scheme \"{}\" for module \"{}\"", scheme, url);
+        eprintln!("Supported schemes: {:#?}", SUPPORTED_URL_SCHEMES);
+        // TODO(bartlomieju): more graceful handling, replace with correct error;
+        return Err(url::ParseError::IdnaError);
```

Comment:
@ry would placeholder `DenoDirError` with message be sufficient for now? During rewrite we can formalize errors.

Reply:
Yes sounds good. That's certainly more intelligible than IdnaError.